cmd/tempo-vulture/main.go

```diff
@@ -258,20 +310,89 @@ func generateRandomInt(min int64, max int64, r *rand.Rand) int64 {
 	return number
 }
 
-func queryTempoAndAnalyze(baseURL string, seed time.Time, traceID string) (traceMetrics, error) {
+func searchTag(client *util.Client, seed time.Time) (traceMetrics, error) {
+	tm := traceMetrics{
+		requested: 1,
+	}
+
+	r := newRand(seed)
+	seedHex := fmt.Sprintf("%016x%016x", r.Int63(), r.Int63())
+	traceID, err := util.HexStringToTraceID(seedHex)
+	if err != nil {
+		return tm, err
+	}
+
+	hexID := util.TraceIDToHexString(traceID)
+
+	// Get the expected
+	expected := constructTraceFromEpoch(seed)
+
+	traceInTraces := func(traceID string, traces []*tempopb.TraceSearchMetadata) bool {
+		for _, t := range traces {
+			equal, err := util.EqualHexStringTraceIDs(t.TraceID, traceID)
+			if err != nil {
+				logger.Error("error comparing trace IDs", zap.Error(err))
+				continue
+			}
+
+			if equal {
+				return true
+
+			}
+		}
+
+		return false
+	}
+
+	logger := logger.With(
+		zap.Int64("seed", seed.Unix()),
+		zap.String("hexID", hexID),
+		zap.Duration("ago", time.Since(seed)),
+	)
+	logger.Info("searching Tempo")
+
+	// Use the search API to find details about the expected trace
+	for _, expectedBatch := range expected.Batches {
```

Comment:
is this loop going to search for every attribute in the trace? maybe we should just choose one randomly and use that?

the two `backoffDuration` config params above are used to control the load that vulture puts on Tempo. if we execute a bunch of queries on each pass we're kind of betraying this config.

Reply:
That's fair.  There are now three backoff config params, but I can see just selecting one of the attributes, so that we know that for each tick of the loops, one call is made to the service and we can make some assumptions about the load characteristics that are added with vulture.